services/beaconchain/src/main/java/tech/pegasys/artemis/services/beaconchain/BeaconChainController.java

```diff
@@ -271,8 +275,12 @@ private void initPreGenesisDepositHandler() {
   }
 
   private void initAttestationPropagationManager() {
-    attestationManager = AttestationManager.create(eventBus, recentChainData);
-    eventChannels.subscribe(SlotEventsChannel.class, attestationManager);
+    final PendingPool<DelayableAttestation> pendingAttestations =
```

Comment:
I had to pull out the creation of pendingAttestations to here so that I can do the subscription to event channel on this level rather than passing event channels to modules below. Not sure whether I should move the initialization of all the objects in AttestationManager, i.e. AttestationProcessor and FutureAttestations to this level, so that they all can follow the dependency injection model.

Reply:
The same thinking applies to the members of BlockPropagationManager below.

Reply:
Makes sense. Will move objects of AttestationPropagationManager and BlockPropagationManager in that direction.